File: API/Features/Reservations/Bookings/ViewModels/BookingListVM.cs
using System;
using System.Collections.Generic;

namespace API.Features.Reservations.Bookings {

    public class BookingListVM {

        public Guid BookingId { get; set; }
        public string BoatName { get; set; }
        public string BoatLength { get; set; }
        public string FromDate { get; set; }
        public string ToDate { get; set; }
        public bool IsConfirmed { get; set; }
        public bool IsDocked { get; set; }
        public bool IsPaid { get; set; }
        public List<BookingPierVM> Piers { get; set; }

    }

}

File: FrontEnd/src/app/features/bookings/classes/view-models/booking-list-vm.ts
import { BookingListPierVM } from './booking-list-pier-vm'

export interface BookingListVM {

    bookingId: string
    boatName: string
    boatLength: string
    fromDate: string
    toDate: string
    stayDuration: number
    piers: BookingListPierVM[]
    joinedPiers: string

}
